libs/langchain/langchain/embeddings/azure_openai.py

```diff
@@ -0,0 +1,149 @@
+"""Azure OpenAI embeddings wrapper."""
+from __future__ import annotations
+
+import os
+import warnings
+from typing import Dict, Optional, Union
+
+from langchain.embeddings.openai import OpenAIEmbeddings
+from langchain.pydantic_v1 import Field, root_validator
+from langchain.utils import get_from_dict_or_env
+from langchain.utils.openai import is_openai_v1
+
+
+class AzureOpenAIEmbeddings(OpenAIEmbeddings):
+    """`Azure OpenAI` Embeddings API."""
+
+    azure_endpoint: Union[str, None] = None
```

Comment:
nit: Optional[str] ? a bit faster to type I think

Reply:
this seems marginally clearer / more readable to me but very subjective